### DISCUSSION TITLE
rxode2 model modification via piping

### DISCUSSION THREAD

**Discussion by dzianismr** [nlmixrdevelopment/RxODE#480] [General]
The post below concerns rxode2 rather than rxode, so it is in the wrong thread here. I reposted it into rxode2 discussion on the topic:
https://github.com/nlmixr2/rxode2/discussions/271

I am very excited to get started with the newly released rxode2 and nlmixr2. I am particularly interested in testing model modification via piping functionality. Do I understand correctly that rxode2 objects can now be modified via piping as well as nlmixr2? What is the right syntaxis?

Lets take a simple 2cmt model (specified below) as an example. Lets say I would like to add inter individual variability to CL parameter (CL = tvCL*exp(eta.CL)). What is the right way of doing this?

I tried first:
rxRename( ode2cmp,  CL = tvCL )
this generates Error: '.data' needs to be a rxUi model

so tried to use functionality of addding a line %>% model(lines, append=TRUE):
ode2cmpt %>% model( lines = "CL = tvCL ", append = TRUE )
and got Error in model.default(., lines = "CL = tvCL ", append = TRUE) : rxode2 does not know how to handle this model statement

Will be really nice to have an example with rxode2 model modification via piping.

ode2cmp <- rxode2({
    #additional variables
    C2 = A2/V2
    C3 = A3/V3
   
    #model
    d/dt(A2) =  - CL*C2 - Q*C2 + Q*C3
    d/dt(A3)  =            Q*C2 - Q*C3
})

**Comment by mattfidler** [comment]
I think this is covered on the `rxode2` discussion board :smile:

**Reply by mattfidler** [reply]
As you already noted at the top (I should have read it more carefully)
